src/validation.cpp

```diff
@@ -5081,7 +5081,9 @@ bool LoadMempool(CTxMemPool& pool)
 
             CAmount amountdelta = nFeeDelta;
             if (amountdelta) {
-                pool.PrioritiseTransaction(tx->GetHash(), amountdelta);
+                if (!pool.PrioritiseTransaction(tx->GetHash(), amountdelta)) {
+                    LogPrintf("PrioritiseTransaction(...) failed. Invalid fee delta?\n");
```

Comment:
This should be literally impossible to hit based on invalid data...?

AFAICT, the only possibility is if the user manages to call `prioritisetransaction` RPC before it's loaded the transaction in question

Reply:
It is possible, but don't take my word for. Take my PoC for it! :)

Proof of concept:

```
$ git checkout master
$ make -C src/ bitcoind
$ xxd -p -r > mempool-signed-integer-overflow.dat << "EOF"
01000000000000003f2d3f3f21f800000000000000000000000000000000
6d697464657363656e64616e00000001000000ec000000003d6a6c000000
000000000000ec9bf601000000000000000000ec9b0001000000000001ff
fffef900000001000000ec0000000000ec9b000001000000000101000100
00000001000000ec000000003d6a6a000000000000000020ec9b000000fa
00
EOF
$ cp mempool-signed-integer-overflow.dat ~/.bitcoin/regtest/mempool.dat
$ UBSAN_OPTIONS="print_stacktrace=1:halt_on_error=1:report_error_type=1" src/bitcoind -regtest
…
txmempool.cpp:839:15: runtime error: signed integer overflow: -7211388903327006720 + -7211353718954917888 cannot be represented in type 'long'
    #0 0x556807f763ec in CTxMemPool::PrioritiseTransaction(uint256 const&, long const&) src/txmempool.cpp:839:15
    #1 0x556807ff3c61 in LoadMempool(CTxMemPool&) src/validation.cpp:5084:22
…
```